metaflow/plugins/kfp/kfp_cli.py

```diff
@@ -282,19 +290,37 @@ def _convert_value(param: parameters.Parameter):
             raise CommandException("Please specify --pipeline-path")
 
         pipeline_path = flow.create_kfp_pipeline_yaml(pipeline_path)
-        obj.echo(
-            "\nDone converting *{name}* to {path}".format(
-                name=current.flow_name, path=pipeline_path
+        if yaml_format == "kfp":
+            pass  # KFP pipeline yaml is created by default
+        elif yaml_format == "argo-workflow":
+            flow.update_kfp_yaml(
+                pipeline_path,
+                [("spec.serviceAccountName", "default-editor")],
             )
-        )
+        elif yaml_format == "argo-workflow-template":
+            flow.update_kfp_yaml(
+                pipeline_path,
+                [
+                    ("spec.serviceAccountName", "default-editor"),
+                    ("kind", "WorkflowTemplate"),
+                    ("status", None),
+                ],
```

Comment:
The conditional logic to modify the workflow yaml would be cleaner to exist in `flow.create_kfp_pipeline_yaml` or rename it to `flow.create_pipeline_yaml`, and it could then modify the `workflow` Dict in memory, and make it much cleaner and easier.

This would also make `kfp run` submitting a workflow to the cluster cleaner and easier to implement. :)